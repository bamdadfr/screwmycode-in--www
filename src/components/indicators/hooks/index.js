export { useIndicatorsComponent } from './use-indicators-component'