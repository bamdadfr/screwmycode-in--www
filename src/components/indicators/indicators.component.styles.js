import styled from 'styled-components'

export const Container = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    margin-bottom: 1em;
    width: 100%;

    * {
        user-select: none;
    }
`

export const Wrapper = styled.div`
    width: 6em;
    margin: 0 0.7em;
    text-align: center;
    display: flex;
    flex-direction: column;
`

export const InputTitle = styled.h2`
    font-size: 0.7em;
    text-transform: uppercase;
`

export const InputValue = styled.input`
    margin-top: 0.7em;
    text-align: center;
    color: ${(props) => props.theme.highlight};
    background: ${(props) => props.theme.background.highlight};
    text-transform: none;
    display: inline-block;
    line-height: 2em;
    border-radius: 3px;
    box-shadow: 0 1px 10px 1px ${(props) => props.theme.shadow.opacity};
    padding: 0 0.5em;
    font-weight: 600;
    font-size: 0.9em;
`