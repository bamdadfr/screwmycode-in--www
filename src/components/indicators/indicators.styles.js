import styled from 'styled-components'

export const StyledContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    margin: 10px 5px -15px 5px;

    * {
        user-select: none;
    }

    > div {
        width: 100px;
        text-align: center;
        display: flex;
        flex-direction: column;

        > input {
            margin-top: 5px;
            width: 100%;
            height: 30px;
            text-align: center;
            border: 0;
            font-size: 1.2em;
            font-weight: 900;
            color: ${(props) => props.theme.colors.white};
            background: transparent;
        }
    }
`