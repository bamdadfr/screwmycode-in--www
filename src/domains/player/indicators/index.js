export { IndicatorsComponent } from './indicators-component'