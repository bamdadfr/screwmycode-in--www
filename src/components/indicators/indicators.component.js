import React from 'react'
import { Container } from './indicators.component.styles'
import { useIndicatorsComponent } from './hooks/use-indicators-component'
import { IndicatorComponent } from './components/indicator/indicator.component'

/**
 * @returns {React.ReactElement} react element
 */
export function IndicatorsComponent () {

    const { percentage, semitones } = useIndicatorsComponent ()

    return (
        <>
            <Container>
                <IndicatorComponent
                    title="percent"
                    metaName="percentage"
                    value={`${percentage} %`}
                />
                <IndicatorComponent
                    title="semitone"
                    metaName="semitones"
                    value={`${semitones} st`}
                />
            </Container>
        </>
    )

}