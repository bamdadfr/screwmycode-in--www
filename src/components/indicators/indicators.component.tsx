import React, {ReactElement} from 'react';

import {IndicatorComponent} from './components/indicator/indicator.component';
import {useIndicatorsComponent} from './hooks/use-indicators-component';
import {Container} from './indicators.component.styles';

/**
 * Component for the indicators
 */
export function IndicatorsComponent(): ReactElement {
  const {percentage, semitones} = useIndicatorsComponent();

  return (
    <Container>
      <IndicatorComponent
        title="percent"
        metaName="percentage"
        value={`${percentage} %`}
      />
      <IndicatorComponent
        title="semitone"
        metaName="semitones"
        value={`${semitones} st`}
      />
    </Container>
  );
}
