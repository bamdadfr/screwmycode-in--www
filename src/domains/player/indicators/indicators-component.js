import React from 'react'
import { StyledContainer } from './indicators-component-styles'
import { useIndicatorsComponent } from './hooks'

/**
 * @returns {React.ReactElement} react element
 */
export function IndicatorsComponent () {

    const { percentage, semitones } = useIndicatorsComponent ()

    return (
        <>
            <StyledContainer>
                <div>
                    <span title="percentage">
                        percent
                    </span>
                    <input
                        disabled
                        aria-label="percentage value"
                        value={percentage}
                    />
                </div>
                <div>
                    <span title="semitones">
                        semitone
                    </span>
                    <input
                        disabled
                        aria-label="semitones value"
                        value={semitones}
                    />
                </div>
            </StyledContainer>
        </>
    )

}