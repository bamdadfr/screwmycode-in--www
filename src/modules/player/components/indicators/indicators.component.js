import React from 'react'
import { Container, Wrapper, InputTitle, InputValue } from './indicators.component.styles'
import { useIndicatorsComponent } from './hooks'

/**
 * @returns {React.ReactElement} react element
 */
export function IndicatorsComponent () {

    const { percentage, semitones } = useIndicatorsComponent ()

    return (
        <>
            <Container>
                <Wrapper>
                    <InputTitle title="percentage">
                        percent
                    </InputTitle>
                    <InputValue
                        disabled
                        aria-label="percentage value"
                        value={`${percentage} %`}
                    />
                </Wrapper>
                <Wrapper>
                    <InputTitle title="semitones">
                        semitone
                    </InputTitle>
                    <InputValue
                        disabled
                        aria-label="semitones value"
                        value={`${semitones} st`}
                    />
                </Wrapper>
            </Container>
        </>
    )

}