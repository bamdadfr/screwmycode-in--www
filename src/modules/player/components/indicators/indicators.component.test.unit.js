import React from 'react'
import { IndicatorsComponent } from './indicators.component'
import { JestInstance } from '../../../../../jest/jest-instance'

const instance = JestInstance (<IndicatorsComponent/>)

it ('should mount and match', () => {

    expect (instance).toBeDefined ()

    expect (instance).toMatchSnapshot ()

})

it ('should have a classname prop from styled component', () => {

    expect (instance.props.className).toBeDefined ()

    expect (instance.props.className).toContain ('sc')

})

it ('should have 2 children', () => {

    expect (instance.children).toHaveLength (2)

})