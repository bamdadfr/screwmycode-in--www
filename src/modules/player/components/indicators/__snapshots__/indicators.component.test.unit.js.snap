// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should mount and match 1`] = `
<div
  className="sc-1ocxull-0 iSsGSx"
>
  <div
    className="sc-1ocxull-1 epeoug"
  >
    <span
      className="sc-1ocxull-2 ewkDmb"
      title="percentage"
    >
      percent
    </span>
    <input
      aria-label="percentage value"
      className="sc-1ocxull-3 ikRRDh"
      disabled={true}
      value="0.0 %"
    />
  </div>
  <div
    className="sc-1ocxull-1 epeoug"
  >
    <span
      className="sc-1ocxull-2 ewkDmb"
      title="semitones"
    >
      semitone
    </span>
    <input
      aria-label="semitones value"
      className="sc-1ocxull-3 ikRRDh"
      disabled={true}
      value="0.0 st"
    />
  </div>
</div>
`;
